File: src/exercise04/tarefa/tarefa02/java/CamaroteSuperior.java
package exercise04.tarefa.tarefa02.java;

public class CamaroteSuperior extends VIP {
    private double additionalValue;

    public CamaroteSuperior(double value, double additionalValueVIP, double additionalValue) {
        super(value, additionalValueVIP);
        this.additionalValue = additionalValue;
    }

    @Override
    public double imprimeValor() {
        double finalPrice = getValue() + getAdditionalValueVIP() + this.additionalValue;
        System.out.println("Ingresso VIP Camarote Superior = R$ " + finalPrice);
        return finalPrice;
        }
}

File: src/exercise04/tarefa/tarefa02/kotlin/CamaroteSuperior.kt
package exercise04.tarefa.tarefa02.kotlin

class CamaroteSuperior(
    value: Double,
    additionalValueVIP: Double,
    private val additionalValue: Double
) :
    VIP(value, additionalValueVIP) {

    override fun imprimeValor(): Double {
        val finalPrice = value + additionalValueVIP + additionalValue
        println("Ingresso VIP Camarote Superior = R$ $finalPrice")
        return finalPrice
    }
}